sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_endpoint_operations.py

```diff
@@ -219,20 +216,29 @@ def invoke(
         :return: The invoked batch deployment job.
         :rtype: BatchJobResource
         """
-        params_override = params_override or []
+        params_override = kwargs.get("params_override", None) or []
+        input = kwargs.get("input", None) # pylint: disable=redefined-builtin
         # Until this bug is resolved https://msdata.visualstudio.com/Vienna/_workitems/edit/1446538
         if deployment_name:
             self._validate_deployment_name(endpoint_name, deployment_name)
 
-        if isinstance(input, Input):
+        if input and isinstance(input, Input):
             if HTTP_PREFIX not in input.path:
                 self._resolve_input(input, os.getcwd())
             # MFE expects a dictionary as input_data that's why we are using
-            # "input_data" as key before parsing it to JobInput
-            params_override.append({EndpointYamlFields.BATCH_JOB_INPUT_DATA: {"input_data": input}})
+            # "UriFolder" or "UriFile" as keys depending on the input type
```

Comment:
Minor: this seems to be unnecessary? As Input objects have a type property?

Reply:
MFE is expecting a dictionary. So we will use the Input type as the "key"

Reply:
Yeah, I get that. What's the point of changing the key from "input_data" to "UriFolder"/"UriFile" though? No need to address this before GA. I just want to make sure I didn't miss anything here.